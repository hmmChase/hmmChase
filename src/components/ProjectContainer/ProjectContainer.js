import React from 'react';
import bitpoll from '../../assets/bitpoll.png';
import monikers from '../../assets/monikers.png';
import ideabox from '../../assets/ideabox.png';
import hmmchase from '../../assets/hmmchase.png';
import palettePicker from '../../assets/palette-picker.png';
import randomQuotes from '../../assets/random-quotes.png';
import Project from '../Project/Project';
import './ProjectContainer.css';

const ProjectContainer = () => (
  <div className='project-container'>
    <Project
      title='IdeaBox'
      img={ideabox}
      desc='IdeaBox is a simple Create, Read, Update, Delete (CRUD) app.  It is implemented three different ways.'
      links={[
        {
          name: 'v1. Vanilla JS',
          gitUrl: 'https://github.com/hmmChase/ideabox-v1/',
          liveUrl: 'https://hmmchase.github.io/ideabox-v1/',
        },
        {
          name: 'v2. React / RESTful',
          gitUrl: 'https://github.com/hmmChase/ideabox-v2/',
          liveUrl: 'https://ideabox-cra-rest.herokuapp.com',
        },
        {
          name: 'v3. SSR React / GraphQL',
          gitUrl: 'https://github.com/hmmChase/ideabox-v3/',
          liveUrl: 'https://ideabox-v3.hmmchase.now.sh',
        },
        {
          name: 'v4. v3 + user accounts',
          gitUrl: 'https://github.com/hmmChase/ideabox-v4/',
          liveUrl: 'https://ideabox-v4.hmmchase.now.sh',
        },
      ]}
    />

    <Project
      title='BitPoll'
      img={bitpoll}
      desc='Bitpoll helps determine consensus within the Bitcoin development community by providing polls restricted to developers of the bitcoin project.'
      links={[
        {
          gitUrl: 'https://github.com/hmmChase/bitpoll/',
          liveUrl: 'https://bit-poll.herokuapp.com',
        },
      ]}
    />

    <Project
      title='Monikers'
      img={monikers}
      desc='A mobile version of the card game Monikers.'
      links={[
        {
          gitUrl: 'https://github.com/hmmChase/monikers/',
          liveUrl: 'https://monikers.herokuapp.com',
        },
      ]}
    />

    <Project
      title='Palette Picker'
      img={palettePicker}
      desc='Palette Picker allows you to generate color schemes and save them in project folders.'
      links={[
        {
          gitUrl: 'https://github.com/hmmChase/palette-picker/',
          liveUrl: 'https://hmm-palette-picker.herokuapp.com',
        },
      ]}
    />

    <Project
      title='Comp Challenges'
      img={
        'https://github.com/hmmChase/cr-comp-challenge-1/blob/master/images/Recreated%20Comp.png'
      }
      desc='Create a responsive layout based on a static comp'
      links={[
        {
          name: 'Challenge 1',
          gitUrl: 'https://github.com/hmmChase/cr-comp-challenge-1/',
          liveUrl: 'https://hmmchase.github.io/cr-comp-challenge-1/',
        },
        {
          name: 'Challenge 2',
          gitUrl: 'https://github.com/hmmChase/cr-comp-challenge-2/',
          liveUrl: 'https://hmmchase.github.io/cr-comp-challenge-2/',
        },
        {
          name: 'Challenge 3',
          gitUrl: 'https://github.com/hmmChase/cr-comp-challenge-3/',
          liveUrl: 'https://hmmchase.github.io/cr-comp-challenge-3/',
        },
      ]}
    />

    <Project
      title='Random Quotes'
      img={randomQuotes}
      desc='Randomly select a quote'
      links={[
        {
          gitUrl: 'https://github.com/hmmChase/Random-Quotes/',
          liveUrl: 'https://rawgit.com/hmmChase/random-quotes/master/',
        },
      ]}
    />

    <Project
      title='hmm... Chase'
      img={hmmchase}
      desc='You are here.'
      links={[
        {
          gitUrl: 'https://github.com/hmmChase/hmmChase/',
          liveUrl: 'https://www.hmmchase.com',
        },
      ]}
    />
  </div>
);

export default ProjectContainer;
