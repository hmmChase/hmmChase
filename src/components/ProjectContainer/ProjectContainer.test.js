import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import ProjectContainer from './ProjectContainer';

describe('ProjectContainer', () => {
  it('matches snapshot', () => {
    const mockProps = {};

    const utils = render(<ProjectContainer {...mockProps} />);

    const container = utils.container.firstChild;

    expect(container).toMatchSnapshot();
  });
});
