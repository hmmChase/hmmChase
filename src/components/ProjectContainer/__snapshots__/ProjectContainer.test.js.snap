// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectContainer matches snapshot 1`] = `
<div
  class="project-container"
>
  <article
    class="project"
  >
    <h3
      class="project-title"
    >
      IdeaBox
    </h3>
    <img
      alt="screenshot"
      class="project-img"
      src="ideabox.png"
    />
    <p
      class="project-description"
    >
      IdeaBox is a simple Create, Read, Update, Delete (CRUD) app.  It is implemented three different ways.
    </p>
    <div
      class="project-link-container"
    >
      <div>
        <p
          class="project-link-name"
        >
          v1. Vanilla JS
        </p>
        <div
          class="project-links"
        >
          <a
            class="project-link"
            href="https://github.com/hmmChase/ideabox-v1"
            rel="noopener noreferrer"
            target="_blank"
          >
            GitHub
          </a>
          <a
            class="project-link"
            href="https://hmmchase.github.io/ideabox-v1/"
            rel="noopener noreferrer"
            target="_blank"
          >
            View Live
          </a>
        </div>
      </div>
      <div>
        <p
          class="project-link-name"
        >
          v2. React / RESTful
        </p>
        <div
          class="project-links"
        >
          <a
            class="project-link"
            href="https://github.com/hmmChase/ideabox-v2"
            rel="noopener noreferrer"
            target="_blank"
          >
            GitHub
          </a>
          <a
            class="project-link"
            href="https://ideabox-cra-rest.herokuapp.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            View Live
          </a>
        </div>
      </div>
      <div>
        <p
          class="project-link-name"
        >
          v3. SSR React / GraphQL
        </p>
        <div
          class="project-links"
        >
          <a
            class="project-link"
            href="https://github.com/hmmChase/ideabox-v3"
            rel="noopener noreferrer"
            target="_blank"
          >
            GitHub
          </a>
          <a
            class="project-link"
            href="https://ideabox-v3.hmmchase.now.sh/"
            rel="noopener noreferrer"
            target="_blank"
          >
            View Live
          </a>
        </div>
      </div>
      <div>
        <p
          class="project-link-name"
        >
          v4. v3 + user accounts
        </p>
        <div
          class="project-links"
        >
          <a
            class="project-link"
            href="https://github.com/hmmChase/ideabox-v4"
            rel="noopener noreferrer"
            target="_blank"
          >
            GitHub
          </a>
          <a
            class="project-link"
            href="https://ideabox-v4.hmmchase.now.sh/"
            rel="noopener noreferrer"
            target="_blank"
          >
            View Live
          </a>
        </div>
      </div>
    </div>
  </article>
  <article
    class="project"
  >
    <h3
      class="project-title"
    >
      BitPoll
    </h3>
    <img
      alt="screenshot"
      class="project-img"
      src="bitpoll.png"
    />
    <p
      class="project-description"
    >
      Bitpoll helps determine consensus within the Bitcoin development community by providing polls restricted to developers of the bitcoin project.
    </p>
    <div
      class="project-link-container"
    >
      <div>
        <div
          class="project-links"
        >
          <a
            class="project-link"
            href="https://github.com/hmmChase/bitpoll"
            rel="noopener noreferrer"
            target="_blank"
          >
            GitHub
          </a>
          <a
            class="project-link"
            href="https://bit-poll.herokuapp.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            View Live
          </a>
        </div>
      </div>
    </div>
  </article>
  <article
    class="project"
  >
    <h3
      class="project-title"
    >
      Monikers
    </h3>
    <img
      alt="screenshot"
      class="project-img"
      src="monikers.png"
    />
    <p
      class="project-description"
    >
      A mobile version of the card game Monikers.
    </p>
    <div
      class="project-link-container"
    >
      <div>
        <div
          class="project-links"
        >
          <a
            class="project-link"
            href="https://github.com/hmmChase/monikers"
            rel="noopener noreferrer"
            target="_blank"
          >
            GitHub
          </a>
          <a
            class="project-link"
            href="https://monikers.herokuapp.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            View Live
          </a>
        </div>
      </div>
    </div>
  </article>
  <article
    class="project"
  >
    <h3
      class="project-title"
    >
      Palette Picker
    </h3>
    <img
      alt="screenshot"
      class="project-img"
      src="palette-picker.png"
    />
    <p
      class="project-description"
    >
      Palette Picker allows you to generate color schemes and save them in project folders.
    </p>
    <div
      class="project-link-container"
    >
      <div>
        <div
          class="project-links"
        >
          <a
            class="project-link"
            href="https://github.com/hmmChase/palette-picker"
            rel="noopener noreferrer"
            target="_blank"
          >
            GitHub
          </a>
          <a
            class="project-link"
            href="https://hmm-palette-picker.herokuapp.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            View Live
          </a>
        </div>
      </div>
    </div>
  </article>
  <article
    class="project"
  >
    <h3
      class="project-title"
    >
      hmm... Chase
    </h3>
    <img
      alt="screenshot"
      class="project-img"
      src="hmmchase.png"
    />
    <p
      class="project-description"
    >
      You are here.
    </p>
    <div
      class="project-link-container"
    >
      <div>
        <div
          class="project-links"
        >
          <a
            class="project-link"
            href="https://github.com/hmmChase/hmmChase"
            rel="noopener noreferrer"
            target="_blank"
          >
            GitHub
          </a>
          <a
            class="project-link"
            href="https://www.youtube.com/watch?v=mz6tAJMVmfM"
            rel="noopener noreferrer"
            target="_blank"
          >
            View Live
          </a>
        </div>
      </div>
    </div>
  </article>
</div>
`;
